data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/AggregateGroup.java

```diff
@@ -42,11 +44,19 @@ Lock getConcludeGroupLock() {
         return concludeGroupLock;
     }
 
+    @Override
+    public void setCustomShouldConclude(Function<Duration, Boolean> shouldConclude) {
+        customShouldConclude = shouldConclude;
+    }
+
     Lock getHandleEventForGroupLock() {
         return handleEventForGroupLock;
     }
 
     boolean shouldConcludeGroup(final Duration groupDuration) {
+        if (customShouldConclude != null) {
+            return customShouldConclude.apply(groupDuration);
```

Comment:
Not necessary for this PR, but this is a nice feature that you've added which could be used at the highest level of the aggregate processor. Perhaps we could have a parameter of `conclude_when`, which takes a conditional expression, and will automatically conclude groups when this condition is true.

Reply:
good suggestion.

Reply:
Thinking more about this, aggregate processor with `duration` has an implicit meaning that conclusion should happen at the end of the duration. Not sure if `conclude_when` makes sense when `duration` is there. May be we should rename `duration` to `conclude_when` and accept duration as one of the possible value for `conclude_when`. What do you think?